### PR TITLE
(2276) Chore: delete activity rake task

### PR DESCRIPTION
**Pull request by mec**
## Changes in this PR

Every report results in a bunch of activities that a developer has to
delete as the application cannot do this yet.

This rake task attempts to aid developers in performing this support
task by running the steps for a given activity database ID.

Some associations are destroyed along with the `Activity` object, others
would be orphaned. I wanted to update the documentation we have on this
and through the process of finding out, I figured we may as well wrap
the knowledge in a rake task.

Developers are encouraged to `think` about what they are doing here as
ultimately this is destructive, but they save a great deal of work on
the console by running the task and interpreting the result before performing 
the destruvtive action.

### CODE REVIEW

**Review by edavey — APPROVED** [review]
Should we have any hard control on activity deletion? e.g. if there are financial data (actual, refund, adjustment, transfer etc) associated with an approved report, should we just refuse to delete the activity? It seems to me that if a mistake has got so far that it's been assured, approved and passed downstream to OECD / IATI then it can't just be "disappeared" and that so other corrective mechanism is needed...

**Comment by mec** [comment]
I agree with you, anything that has been assured and approved should never be deleted - but I am not convinced we are in that world just yet:

If we imagine that we cannot delete an activity that has approved data in the current world we would have to delete the errant activities before the assurance is complete i.e. the final step in assurance is to delete the errant acticities. I think this would place too great a burden on the support team becoming the gate keepers to approval, which could only happen once the deleting has taken place.

My view is that right now, we have to rely on the BEIS folks supplying the correct list of activities for deletion and an developer running the task only has to deal with making sure the ID used is the correct one and not that the activity itself is correct to be deleted, which should not be our problem.

My hope is that as the service matures we will see less activities to be deleted or even better, move deletion into the application where constraints like the one proposed could be enforced - I think there is more thought to be put into exactly what an acitivity that can be deleted would look like, there may be more than we have considered. In that instance, the assurance team could delete the errant activity before it ever got approve - though I suspect mistakes will still get through!

We have even previously considered that activites are NEVER deleted and merely marked as 'errors' and not shown to anyone unless explicitally requested - unfortunately that thinking and work never made it to the top of the priority order! :(

As you have given approval I'll assume this is a thinking outloud comment and go ahead with this PR - but let me know if you think we should add something, or at least flag something in this task?

**Pull request merged**
